sql/item_inetfunc.cc

```diff
@@ -827,7 +827,7 @@ size_t Inet6::to_string(char *dst, size_t dstsize) const
       //
       // If it is not the last field, append closing ':'.
 
-      p += sprintf(p, "%x", ipv6_words[i]);
+      p += snprintf(p, sizeof(p), "%x", ipv6_words[i]);
```

Comment:
p isn't a character array.

Reply:
@grooverdan `sprintf` also accepts character pointer by definition same as `snprintf. Just trying to understand why would array is required?

Reply:
```
#include <stdio.h>
#include <stdlib.h>

int main()
{ char *p, a[300];
  p= malloc(20);
  printf("sizeof(p) = %zu, sizeof(a) = %zu\n", sizeof(p), sizeof(a));
  return 0;
}
```

```
$ /tmp/main
sizeof(p) = 8, sizeof(a) = 300
```

See that sizeof isn't the allocated size except with where its known at compile time.

Reply:
Here what I think...
So, the problem is pointer's size will be different than that of Array. Pointer's size will be calculated at the time of compiling based on OS, but Array could be of any value. But as `p` is just a pointer, `sizeof(p)` is not going to be correct. 
Is that correct understanding?

Reply:
Reverted back to sprintf function